core/embed/rust/src/ui/component/text/layout.rs

```diff
@@ -537,13 +600,30 @@ impl Span {
 
         let mut span_width = 0;
         let mut found_any_whitespace = false;
+        let mut chunks_wider_chars = 0;
 
         let mut char_indices_iter = text.char_indices().peekable();
         // Iterating manually because we need a reference to the iterator inside the
         // loop.
         while let Some((i, ch)) = char_indices_iter.next() {
             let char_width = text_font.char_width(ch);
 
+            // When there is a set chunk size and we reach it,
+            // adjust the line advances and return the line.
+            if let Some(chunkify_config) = chunks {
+                if i == chunkify_config.chunk_size {
+                    line.advance.y = 0;
+                    // Decreasing the offset for each wider character in the chunk
+                    line.advance.x += chunkify_config.x_offset - chunks_wider_chars;
+                    return line;
```

Comment:
I wonder if this can't break line-counting somewhere?

Reply:
The tests seem to be alright, it works even with longer addresses that paginate (Cardano ones)